airflow/stats.py

```diff
@@ -291,10 +315,11 @@ def wrapper(self, stat=None, *args, tags=None, **kwargs):
         if self.influxdb_tags_enabled:
             if stat is not None and tags is not None:
                 for k, v in tags.items():
-                    if all((c not in [",", "="] for c in v + k)):
-                        stat += f",{k}={v}"
-                    else:
-                        log.error("Dropping invalid tag: %s=%s.", k, v)
+                    if self.metric_tags_validator.test(k):
```

Comment:
Why do we need to check if the tag is filtered here?

Reply:
@hussein-awala 

Aren't we only concatenating metric tags that are allowed? (i.e. not disabled / not blocked)

If it's test(k) returns True, we will want to publish that metric tag

Reply:
My bad, I thought that this filter:
```python
            tags_list = [
                f"{key}:{value}" for key, value in tags.items() if self.metric_tags_validator.test(key)
            ]
```
is used for the two clients datadog and influxdb, but just found that is just used for datadog, and `if self.metric_tags_validator.test(k):` is used for influxdb

Reply:
Yes - influxdb integration just adds the tag key:value pairs to the metric name with a comma-delimited standard, instead of having a separate Statsd client implementation that takes tags as a separate parameter.